cs/src/core/FasterLog/FasterLog.cs

```diff
@@ -120,12 +120,21 @@ public sealed class FasterLog : IDisposable
         /// Used to determine disposability of log
         /// </summary>
         internal int logRefCount = 1;
-        
+
         /// <summary>
         /// Create new log instance
         /// </summary>
         /// <param name="logSettings">Log settings</param>
         public FasterLog(FasterLogSettings logSettings)
+            : this(logSettings, logSettings.TryRecoverLatest)
+        { }
+
+        /// <summary>
+        /// Create new log instance
+        /// </summary>
+        /// <param name="logSettings">Log settings</param>
+        /// <param name="syncRecover">Recover synchronously</param>
+        private FasterLog(FasterLogSettings logSettings, bool syncRecover)
```

Comment:
why not have syncRecover = true to cover the earlier case as well?

Reply:
We want public constructor to be clean with only one argument, the settings.

Reply:
Ok